editor/AmbientSoundPattern.cpp

```diff
@@ -224,11 +224,9 @@ void CAmbientSoundPattern::OnASPNewElement() {
     mem_free(asp->sounds);
 
     asp->sounds = new_sounds;
-    asp->sounds[s] = new_element;
+    asp->sounds[asp->num_sounds++] = new_element;
 
-    asp->num_sounds++;
-
-    m_current_element = s;
+    m_current_element = asp->num_sounds;
```

Comment:
we incremented num_sounds just before, so now `asp->num_sounds` is `s+1`. This should be `m_current_element = asp->num_sounds-1;` ?